opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/LuceneQuery.java

```diff
@@ -209,7 +214,16 @@ private ExprValue cast(FunctionExpression castFunction) {
           .put(
               BuiltinFunctionName.CAST_TO_DATE.getName(),
               expr -> {
-                if (expr.type().equals(ExprCoreType.STRING)) {
+                if (expr.type().equals(ExprCoreType.STRING)
```

Comment:
The code is duplicate as 235-244 and 253-262, could u simplify.

Reply:
Regarding to parse string as ExprDateValue, we can extend supported formatted in ExprDataValue and update user doc. 
The literay format data value in SQL not necessary follow OpenSearch index dataformat.

for instance, if SQL only support date in format 2024-01-03, it is also fine. unless we can make sure the literary value is rewrite as 03-Jan-24 in DSL query.
```
POST _plugins/_sql
{
    "query":"select release_date from testindex where release_date='03-Jan-24'"
}


POST _plugins/_sql
{
    "query":"select release_date from testindex where release_date='2024-01-03'"
}
```

Reply:
Does it mean its ok to not support the below query? 
```
POST _plugins/_sql
{
    "query":"select release_date from testindex where release_date='03-Jan-24'"
}
```
Instead, when field mapping format has `dd-MM-yy`, we would still take '2024-01-03' string as part of query and do formatting to match `dd-MM-yy` just before sending request in DSL query?

Reply:
Too many code rebases, haven't catch up the latest commit. But if the only target is to support two-digits year, maybe you can refer my poc [here](https://github.com/LantaoJin/search-plugins-sql/commit/265be074cb22a7f70ed730568e5f7f1f44c043f0), it armed to fix the issue [[BUG] str_to_date does not work with two-digits year](https://github.com/opensearch-project/sql/issues/2741)